glow/rate_limiter_test.go

```diff
@@ -0,0 +1,84 @@
+package glow
+
+import (
+	"testing"
+	"time"
+)
+
+func TestRateLimiterSingle(t *testing.T) {
+	rl := NewRateLimiter(1, 50*time.Millisecond)
+
+	ticker := time.NewTicker(20 * time.Millisecond)
+	defer ticker.Stop()
+
+	count := 0
+	allowed := 0
+	for _ = range ticker.C {
+		if rl.Allow() {
+			allowed++
+		}
+		count++
+		if count == 4 {
+			break
+		}
+	}
+
+	t.Logf("allowed %v", allowed)
+
+	if allowed != 2 {
+		t.Errorf("Single test failed, wanted %v got %v", 2, allowed)
+	}
+}
+
+// Basic multithreaded RateLimiter test
+func TestRateLimiterMultithreaded(t *testing.T) {
+
+	maxcalls := 5
+	maxdur := 100 * time.Millisecond
+	testcalls := 19
+	testint := 15 * time.Millisecond
```

Comment:
I would probably create a loop that tries an assortment of different combinations of values for these values

Reply:
just gives better confidence that edge cases are handled.

Reply:
Makes sense, single case tests can be subconciously hardcoded to pass. Thanks for the reminder!